### PR TITLE
Fixes VV Make Space Ninja verb

### LINKED ISSUE #32895: "Make space ninja" through VV is broken.
Selecting it does nothing.

### PR DESCRIPTION
**Pull request by JJRcop**
:cl: JJRcop
admin: Fixed the Make space ninja verb.
/:cl:

Fixes #32895

### CODE REVIEW

**Comment by JJRcop** [comment]
Remove the administration tag

**Comment by PKPenguin321** [comment]
Give it a confirm/cancel prompt too if it doesn't have one already

**Comment by JJRcop** [comment]
If you want to do that then delete this line: https://github.com/tgstation/tgstation/blob/080c323ec2c88deb8da9beb0210f3e9fda5f98e9/code/modules/mob/mob.dm#L966

File: code/datums/antagonists/ninja.dm [hunk 1]
```diff
@@ -58,7 +58,7 @@
 				var/is_bad_guy = possible_targets[M]
 				possible_targets.Cut(index,index+1)
 
-				if(is_bad_guy ^ helping_station)			//kill (good-ninja + bad-guy or bad-ninja + good-guy)
+				if(is_bad_guy && !helping_station)			//kill (good-ninja + bad-guy or bad-ninja + good-guy)
```
Comment (RemieRichards) [inline code/datums/antagonists/ninja.dm]: Revert these.

Comment (JJRcop) [reply code/datums/antagonists/ninja.dm]: I wish I read the comment first, thanks.

File: code/datums/datumvars.dm [hunk 2]
```diff
@@ -748,6 +748,28 @@
 			src.give_disease(M)
 			href_list["datumrefresh"] = href_list["give_spell"]
 
+		else if(href_list["ninja"])
+			if(!check_rights(R_FUN))
+				return
+
+			var/mob/M = locate(href_list["ninja"])
```
Comment (lzimann) [inline code/datums/datumvars.dm]: this locate needs to have a scope, you can use either `mob_list` or typecast it to a human right away, then you can use `carbon_list`.

File: code/datums/datumvars.dm [hunk 3]
```diff
@@ -748,6 +748,28 @@
...
+			if(tgalert(usr, "Are you sure you want to make [M] into a ninja?", "Confirmation", "Yes", "No") == "No")
+				return
+
+			if(!M.mind)
+				M.mind_initialize()
+
+			var/datum/antagonist/ninja/hiyah = M.mind.has_antag_datum(/datum/antagonist/ninja)
+			if(!hiyah)
+				hiyah = add_ninja(M)
+			if(hiyah)
```
Comment (lzimann) [inline code/datums/datumvars.dm]: `else`

Comment (JJRcop) [reply code/datums/datumvars.dm]: add_ninja returns a /datum/antagonist/ninja, or if it faiks, it doesn't, thia behavior is intended and an if else will break it

Comment (lzimann) [reply code/datums/datumvars.dm]: right, i missed it, my bad.

**Review by Cyberboss — DISMISSED** [review]
Stuff lzi said

**Comment by JJRcop** [comment]
Sorry I let it go stale

**Pull request merged**